lib/ShareProvider.js

```diff
@@ -146,6 +150,32 @@ const Overlay = {
     if (window.SocialShare.shareButton) {
       window.SocialShare.shareButton.setAttribute("hidden", "true");
     }
+
+    // monkeypatch SocialActivationListener.receiveMessage
+    window.SocialActivationListener.originalReceiveMessage = window.SocialActivationListener.receiveMessage;
+    window.SocialActivationListener.receiveMessage = function(aMessage) {
+      // NOTE: This is a patched version of SocialActivationListener.receiveMessage
+      // that doesn't install the shareplane button to the toolbar when a new provider
+      // is activated.
+      // original code is at mozilla-central/source/browser/base/content/browser-social.js
+      let data = aMessage.json;
+      data.window = window;
+      const {
+        gBrowser,
+        Social,
+        SocialShare
+      } = window;
+
+      Social.installProvider(data, manifest => {
+        Social.activateFromOrigin(manifest.origin, provider => {
+          if (provider.postActivationURL) {
+            // if activated from an open share panel, we load the landing page in
+            // a background tab
+            gBrowser.loadOneTab(provider.postActivationURL, {inBackground: SocialShare.panel.state === "open"});
```

Comment:
drop the SocialShare.panel check (just use true or false as you prefer) then r+ from me.